test/e2e/scheduling/priorities.go

```diff
@@ -0,0 +1,713 @@
+		Expect(err).NotTo(HaveOccurred())
+		By("Ensuring each pod is running")
+		result := map[string]bool{}
+		for _, pod := range pods.Items {
+			if pod.DeletionTimestamp != nil {
+				continue
+			}
+			err = f.WaitForPodRunning(pod.Name)
+			Expect(err).NotTo(HaveOccurred())
+			result[pod.Spec.NodeName] = true
+		}
+		By("Verify the pods were scheduled to each node")
+		if len(nodeList.Items) != len(result) {
+			framework.Failf("Pods are not spread to each node")
+		}
+	})
+
+	It("Pods Belong to same StatefulSet should spread to different node", func() {
+		By("Create a pod for each node to make the nodes have almost same cpu/mem usage")
+
+		// make the nodes have balanced cpu,mem usage
+		createBalancedPodForNodes(f, cs, ns, nodeList.Items, &v1.ResourceRequirements{}, 0.6)
+
+		By("Create a StatefulSet, with pods No. same with schedualble node NO. pods should schedule to each node")
+		ss := framework.NewStatefulSet("scheduler-priority-selector-spreading", f.Namespace.Name, "test", int32(len(nodeList.Items)), nil, nil, map[string]string{"statefulset": "ss1"})
+		_, err := cs.Apps().StatefulSets(ns).Create(ss)
+		Expect(err).NotTo(HaveOccurred())
+
+		pods, err := framework.PodsCreatedByLabel(f.ClientSet, f.Namespace.Name, "scheduler-priority-selector-spreading", int32(len(nodeList.Items)), labels.SelectorFromSet(labels.Set(map[string]string{"statefulset": "ss1"})))
+		Expect(err).NotTo(HaveOccurred())
+		By("Ensuring each pod is running")
+
+		result := map[string]bool{}
+		for _, pod := range pods.Items {
+			if pod.DeletionTimestamp != nil {
+				continue
+			}
+			err = f.WaitForPodRunning(pod.Name)
+			Expect(err).NotTo(HaveOccurred())
+			result[pod.Spec.NodeName] = true
+		}
+		By("Verify the pods were scheduled to each node")
+		if len(nodeList.Items) != len(result) {
+			framework.Failf("Pods are not spread to each node")
+		}
+	})
+
+	It("Pod should be prefer scheduled to node that satisify the NodeAffinity", func() {
+		nodeName := GetNodeThatCanRunPod(f)
+		By("Trying to apply a label on the found node.")
+		k := fmt.Sprintf("kubernetes.io/e2e-%s", "node-topologyKey")
+		v := "topologyvalue"
+		framework.AddOrUpdateLabelOnNode(cs, nodeName, k, v)
+		framework.ExpectNodeHasLabel(cs, nodeName, k, v)
+		defer framework.RemoveLabelOffNode(cs, nodeName, k)
+
+		// make the nodes have balanced cpu,mem usage ratio
+		createBalancedPodForNodes(f, cs, ns, nodeList.Items, podRequestedResource, 0.6)
+		By("Trying to relaunch the pod, now with labels.")
+		labelPodName := "pod-with-node-affinity"
+		pod := createPausePod(f, pausePodConfig{
+			Name: labelPodName,
+			Affinity: &v1.Affinity{
+				NodeAffinity: &v1.NodeAffinity{
+					PreferredDuringSchedulingIgnoredDuringExecution: []v1.PreferredSchedulingTerm{
+						{
+							Preference: v1.NodeSelectorTerm{
```

Comment:
@k82cn No, just add two item to match the affinity, but we can using one selector if you don't like.